src/equations/compressible_euler_multicomponent_1d.jl

```diff
@@ -0,0 +1,491 @@
+  rhok_avg    = SVector{ncomponents(equations), real(equations)}(0.5 * (u_ll[i+2] + u_rr[i+2]) for i in eachcomponent(equations))
+
+  # Iterating over all partial densities
+  rho_ll      = density(u_ll, equations)
+  rho_rr      = density(u_rr, equations)
+
+  gamma_ll = totalgamma(u_ll, equations)
+  gamma_rr = totalgamma(u_rr, equations)
+
+  # extract velocities
+  v1_ll       = rho_v1_ll/rho_ll
+  v1_rr       = rho_v1_rr/rho_rr
+  v1_avg      = 0.5 * (v1_ll + v1_rr)
+  v1_square   = 0.5 * (v1_ll^2 + v1_rr^2)
+  v_sum       = v1_avg
+
+  enth      = zero(v_sum)
+  help1_ll  = zero(v1_ll)
+  help1_rr  = zero(v1_rr)
+
+  for i in eachcomponent(equations)
+    enth      += rhok_avg[i] * gas_constants[i]
+    help1_ll  += u_ll[i+2] * cv[i]
+    help1_rr  += u_rr[i+2] * cv[i]
+  end
+ 
+  T_ll        = (rho_e_ll - 0.5 * rho_ll * (v1_ll^2)) / help1_ll
+  T_rr        = (rho_e_rr - 0.5 * rho_rr * (v1_rr^2)) / help1_rr
+  T           = 0.5 * (1.0/T_ll + 1.0/T_rr)
+  T_log       = ln_mean(1.0/T_ll, 1.0/T_rr)
+
+  # Calculate fluxes depending on orientation
+  help1       = zero(T_ll)
+  help2       = zero(T_rr)
+  
+  f_rho       = SVector{ncomponents(equations), real(equations)}(rhok_mean[i]*v1_avg for i in eachcomponent(equations))
+  for i in eachcomponent(equations)
+    help1     += f_rho[i] * cv[i]
+    help2     += f_rho[i]
+  end
+  f1 = (help2) * v1_avg + enth/T
+  f2 = (help1)/T_log - 0.5 * (v1_square) * (help2) + v1_avg * f1
+  
+  f_other  = SVector{2, real(equations)}(f1, f2)
+
+  return vcat(f_other, f_rho)
+end
+
+
+function flux_lax_friedrichs(u_ll, u_rr, orientation, equations::CompressibleEulerMulticomponentEquations1D)
+  # Calculate primitive variables and speed of sound
+  rho_v1_ll, rho_e_ll = u_ll
+  rho_v1_rr, rho_e_rr = u_rr
+
+  rho_ll    = density(u_ll, equations)
+  rho_rr    = density(u_rr, equations)
+  gamma_ll  = totalgamma(u_ll, equations)
+  gamma_rr  = totalgamma(u_rr, equations)
+
+  v1_ll     = rho_v1_ll / rho_ll
+  v1_rr     = rho_v1_rr / rho_rr
+  v_mag_ll  = sqrt(v1_ll^2)
+  v_mag_rr  = sqrt(v1_rr^2)
+
+  p_ll = (gamma_ll - 1) * (rho_e_ll - 1/2 * rho_ll * v_mag_ll^2)
+  c_ll = sqrt(gamma_ll * p_ll / rho_ll)
+  p_rr = (gamma_rr - 1) * (rho_e_rr - 1/2 * rho_rr * v_mag_rr^2)
+  c_rr = sqrt(gamma_rr * p_rr / rho_rr)
+
+  # Obtain left and right fluxes
+  f_ll = flux(u_ll, orientation, equations)
+  f_rr = flux(u_rr, orientation, equations)
+
+  λ_max = max(v_mag_ll, v_mag_rr) + max(c_ll, c_rr)
+
+  f  = SVector{nvariables(equations), real(equations)}(1/2 * (f_ll[i] + f_rr[i]) - 1/2 * λ_max * (u_rr[i] - u_ll[i]) for i = 1:nvariables(equations))
+
+  return f
+end
+
+
+@inline function max_abs_speeds(u, equations::CompressibleEulerMulticomponentEquations1D)
+  rho_v1, rho_e = u
+
+  rho   = density(u, equations)
+  v1    = rho_v1 / rho
+
+  gamma = totalgamma(u, equations)
+  p     = (gamma - 1) * (rho_e - 1/2 * rho * (v1^2))
+  c     = sqrt(gamma * p / rho)
+
+  return abs(v1) + c
```

Comment:
```suggestion
  return (abs(v1) + c, )
```
We also return a `Tuple` for other systems in 1D.